eth2util/eth2exp/attagg.go

```diff
@@ -17,200 +17,103 @@ package eth2exp
 
 import (
 	"context"
-	"encoding/binary"
 	"encoding/hex"
 	"encoding/json"
 	"fmt"
 	"strconv"
 	"strings"
 
-	eth2client "github.com/attestantio/go-eth2-client"
 	eth2p0 "github.com/attestantio/go-eth2-client/spec/phase0"
-	"github.com/minio/sha256-simd"
 
 	"github.com/obolnetwork/charon/app/errors"
 )
 
-type eth2Provider interface {
-	eth2client.SpecProvider
-}
-
-// BeaconCommitteeSubscriptionsSubmitterV2 is the interface for submitting beacon committee subnet subscription requests.
+// BeaconCommitteeSelectionAggregator is the interface for aggregating beacon committee selection proofs in a DVT cluster.
 // TODO(dhruv): Should be removed once it is supported by go-eth2-client.
-type BeaconCommitteeSubscriptionsSubmitterV2 interface {
-	// SubmitBeaconCommitteeSubscriptionsV2 subscribes to beacon committees.
-	SubmitBeaconCommitteeSubscriptionsV2(ctx context.Context, subscriptions []*BeaconCommitteeSubscription) ([]*BeaconCommitteeSubscriptionResponse, error)
+type BeaconCommitteeSelectionAggregator interface {
```

Comment:
this can be `BeaconCommitteeSelectionSubmitter` or `BeaconCommitteeSelectionAggregateSubmitter`

Reply:
It isn't a pure submitter nor a pure provider. You submit a partial selection, and it returns an aggregated selection. Eitherway, this is a name that Jim will choose for his library and isn't in the official spec or in our code base.